src/main/java/touro/chess/Board.java

```diff
@@ -4,14 +4,134 @@
 
     private static final int COLUMNS = 8;
     private static final int ROWS = 8;
+    private Square[][] squares = new Square[COLUMNS][ROWS];
+
+    /**
+     * Creates an empty board
+     */
+    public Board()
+    {
+        for(int column = 0; column < COLUMNS; column++) {
+            for (int row = 0; row < ROWS; row++) {
+                squares[column][row] = new Square(new Location(row, column));
+            }
+        }
+    }
 
     /**
      * Verify that a Move is legal on this Board.
      * @param move
      * @return true if the Move is legal, otherwise false.
      */
     public boolean isLegal(Move move) {
+        int currentRow = move.getFrom().getRow();
+        int currentColumn = move.getFrom().getColumn();
+        int destinationRow = move.getTo().getRow();
+        int destinationColumn = move.getTo().getColumn();
+
+        if (currentColumn < 0 || currentColumn > 7
```

Comment:
I would probably make these constants...

Reply:
You should make an isInBoard() method.